\documentclass[10pt,draft,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage[nopatch=eqnum]{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\usepackage{setspace}
\doublespacing

% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

%% END MACROS SECTION

\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Bisulphite sequencing in the presence of cytosine-conversion errors} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Thomas James Ellis\textsuperscript{1},
Viktoria Nizhynska\textsuperscript{1},
Rahul Pisupati\textsuperscript{1},
Almudena Moll\'a-Morales\textsuperscript{1},
Magnus Nordborg\textsuperscript{1*}
\\
\bigskip
\textbf{1} Gregor Mendel Institute, Austrian Academy of Sciences, Vienna BioCenter, Vienna, Austria.
\\
\bigskip

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* magnus.nordborg@gmi.oeaw.ac.at

\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
Bisulphite treatment of DNA converts unmethylated cytosines to thymine, and is a common method to infer the methylation status of cytosines when coupled with sequencing.
`Tagmentation' approaches to bisulphite sequencing use a transposase to simultaneously make double-stranded breaks and ligate adaptors to the resulting fragments, allowing for higher throughput with less starting material.
However, it has also been noted that certain tagmentation protocols have an unusually high number unmethylated cytosines that are not converted to thymine.
Here we describe this phenomenon in detail, and find that results are consistent with single strand nicks by the transposase, followed by strand displacement of part or all of the DNA fragment, leading to erroneous incorporation of methylated cytosines.
Nevertheless we show that these errors can be accounted for in downstream analysis and need not impede biological conclusions.
We provide a Python package to allow users to implement this framework.
Ultimately the additional effort of accounting for errors must be traded off against the scalability of the protocol in planning experiments.


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
\section*{Author summary}

Cytosines are frequently modified by the addition of a methyl group, and this methylation is associated with important regulatory functions in the genome.
The most common way to quantify cytosine methylation is by treatment with sodium bisulphite, which converts unmethylated cytosines to thymine.
However this process can be error prone, causing estimated methylation to be biased or noisy.
Here we investigate this pattern in detail for a bisulphite treatment protocol optimised for small quantities of DNA.
The patterns are consistent with mistakes by a key enzyme in the protocol that cause whole fragments to become methylated.
We describe how these mistakes can be accounted for in a way that does not bias biological conclusions.

\linenumbers

\section*{Introduction}

Cytosine methylation is a common epigenetic mark associated with heterochromatin and transcriptional silencing.
A common approach to assay methylation is to treat DNA with sodium bisulphite to convert unmethylated cytosines to uracil that are in turn converted to thymine in a subsequent PCR step \cite{clark1994high}.
Methylated cytosines remain intact.
Following sequencing of the resulting DNA fragments, reads are aligned to a reference genome.
The methylation status at each site is determined based on how many thymines and cytosines align to each genomic cytosine position.
This gives a quantitative estimate of the proportion of cells in which each cytosine is methylated.

A major limitation of bisulphite treatment is that the process itself damages DNA, meaning that substantial amounts of starting material are required for adequate results.
Early studies provided the first genome-wide surveys of methylation profiles, but relied on sonication of DNA followed by a delicate adaptor-ligation step (e.g. \cite{meissner2005reduced, cokus2008shotgun, lister2009human}).
Aiming to make the procedure more easily scalable to high-throughput samples, `tagmentation' approaches use a Tn5 transposase to simultaneously cut the DNA and ligate adaptors and PCR primers to the fragments \cite{wang2013tagmentation}.
In the original protocol, the transposase is loaded with a single adaptor, and the second adaptor is annealed by by oligo replacement, followed by repair of the nine-base-pair single-stranded gap left by the transposase \cite{adey2012ultra}.
This allows for several orders of magnitude less starting material than sonication-based protocols, but the oligo-replacement step remains the most challenging step.
Lu \textit{et al.} \cite{lu2015improved} modified this approach to load the transposome with two methylated adaptors, and to replace the oligo-replacement and gap repair steps with a single strand displacement reaction by a \textit{Bst} polymerase.
The strand displacement step simultaneously fills the nine-base-pair gap and replaces the 3`-adaptor with a methylated oligonucleotide containing the
reverse sequencing primer (see also \cite{weichenhan2019generation, suzuki2018whole}).
In contrast to the previous approach, gap repair is carried out using dNTPs with 5-methyl-dCTPs instead of dCTPs.
Since these gap regions are removed as part of the adaptors during bioinformatic processing, these methyl-dCTPs should not in principle affect methylation of genomic DNA.
Using two adaptors also has the advantage that the original and complementary strands generated by PCR can be distinguished by read-alignment software, effectively doubling coverage. 
The stand-displacement tagmentation protocol thus has the potential to yield more information from less starting material.

However, accurate inference of methylation states requires that unmethylated cytosines be reliably converted to thymines.
A standard quality-control step is to include unmethylated control DNA, and to estimate the conversion rate as the proportion of cytosines that are sequenced as thymines.
In the ideal case, all unmethylated cytosines should be converted to thymine.
Lu \textit{et al.} \cite{lu2015improved} and Suzuki \textit{et al.} \cite{suzuki2018whole} reported that 1-2\% of reads generated by a strand-displacement protocol appeared to be entirely unconverted.
Lu \textit{et al.} \cite{lu2015improved} hypothesised that this phenomenon is due to rare single-strand nicks in the 5` adaptors that serve as a start site for extension by the \textit{Bst} polymerase, leading to strand displacement over the whole fragment.
Because the protocol uses 5-methyl-dCTPs, this would cause the whole strand to become methylated, potentially biasing or obscuring real methylation patterns.
Lu \textit{et al.} \cite{lu2015improved} and Suzuki \textit{et al.} \cite{suzuki2018whole} filtered out suspicious reads, but do not discuss the issue further.

Here we address this issue by providing a detailed description of non-conversion errors using strand-displacement-based tagmentation.
We first describe patterns of non-conversion within reads and across the genome and a possible mechanism for their source.
We describe a statistical framework to account for non-conversion errors to quantify and classify real cytosine methylation, and demonstrate that methylation levels can be inferred in the presence of non-conversion errors.

\section*{Results and Discussion}

\subsection*{Non-conversion affects all or part of a read}

We investigated how and where cytosine non-conversion errors occur in \textit{Arabidopsis thaliana}, \textit{Drosophila melanogaster} and phage $\lambda$.
Plant chloroplasts, and the genomes of fruit flies and this virus lack cytosine methylation, so any unconverted cytosines we observe must be erroneous.
We only examined the first half of the \textit{A.~thaliana} chloroplast, because the second half shows strong homology with the autosomes.
We used a modified version of the strand-displacement protocol described by Weichenhan \textit{et al.} \cite{weichenhan2019generation} but varied the concentration of transposase and number of PCR cycles to identify possible artefacts arising from these aspects of the protocol.
Overall cytosine non-conversion rates were very high, ranging from a few percent up to more than 17\% (Fig~\ref{fig:reads}A).
For comparison, methylation on chloroplasts in \textit{A.~thaliana} is typically a fraction of a percent \cite{zhang2018large}.
There was substantial variation between libraries, but there was no consistent effect of species, transposase concentration, or number of PCR cycles.
Since we used the same DNA preparation for each library, this is technical variation between libraries rather than biological variation.
In particular, we note that the highest non-conversion rates for phage $\lambda$ were prepared as part of the same libraries as the samples of \textit{D.~melanogaster} with the highest non-conversion rates, indicating that the effect is due primarily to differences between libraries.
High cytosine non-conversion rates do not seem to be constrained to any species, and there is substantial stochasticity between samples.

\begin{figure}
  % \begin{center}
    \includegraphics{figure1.eps}
    \caption{
        {\bf Cytosine non-conversion within and between reads on unmethylated DNA from \textit{A.~thaliana}, \textit{D.~melanogaster} and Lambda phage}.
        For \textit{A.~thaliana}, data are shown for reads aligning to the first 75kb of chloroplast.
        (A) Overall cytosine non-conversion rates in libraries prepared using different relative concentrations of Tn5 transposase to DNA, and 13 or 15 PCR cycles.
        (B) The distribution of the proportion of unconverted cytosines across reads. For clarity, reads with no unconverted cytosines are not shown.
        (C) The probability of observing an unconverted cytosine at each base0pair position within reads each of the four strands generated by bisulphite libraries (OT=original top strand; OB=original bottom; CTOT=complementary to original top; CTOB=complementary to original bottom) averaged over libraries.
    }
    \label{fig:reads}
  % \end{center}
\end{figure}

To investigate the phenomenon further we examined the distribution of cytosine non-conversion within and between sequencing reads.
Reads containing at least one cytosine fell into three categories.
The majority of reads (71\%, on average) showed no unconverted cytosines at all.
Remaining reads were roughly evenly split between those where every cytosine on a read was unconverted, and those where only a proportion of cytosines were unconverted (Fig~\ref{fig:reads}B).
The bimodal distribution of reads indicates that two distinct processes contribute to errant cytosine non-conversion.
These observations contrast with previous reports of errant non-conversion in mammals which found that only 1-2\% of reads being affected, and that when errors did occur, they occurred across the entire read \cite{lu2015improved, suzuki2018whole}.

We found that non-conversion errors in partly-methylated reads were more likely to occur towards the 3` than the 5` end of the read (Fig~\ref{fig:reads}C).
This may be partly due to the general tendency of sequence quality to decline along a read.
However, this is unlikely to be the main cause in this case because Phred scores are consistently high across reads.
It may also be that the presence of fully, and densely methylated cytosines in the adaptor sequences interfere with cytosine conversion.
This is also unlikely, because in this case we would expect increased non-conversion at both ends of the read.
This pattern must be due to reads with only partial non-conversion, since fully unconverted reads would increase non-conversion uniformly across the whole read.

There is also some evidence that non-conversion errors depend on strand.
Bisulphite libraries typically distinguish between four strands: the original top and bottom strands (OT, OB), and the strands complementary to the original (CTOT) and bottom (CTOB) strands that are synthesised during PCR.
In the \emph{A.~thaliana} and phage $\lambda$ samples there is a slight tendency for the top strands to have increased errors compared to the bottom strands, and for the complementary strands to have increased errors compared to the corresponding original strands.
In the \emph{D.~melanogaster} samples there also appears to be a periodicity in the probabilities of non-conversion every five bases on the OB and CTOB strands only.
The reasons for this apparent strand-specificity are unclear.

\subsection*{Non-conversion errors vary across the genome}

Taking the average cytosine non-conversion rate on control DNA gives a point estimate of the genome-wide non-conversion rate, but this might mask variation in error rates between regions of the genome.
To investigate this we estimated non-conversion rates in 150~bp windows across the \emph{A.~thaliana} chloroplast, corresponding to the approximate size of one loop of a nucleosome.
We found as much as five-fold variation in cytosine non-conversion between windows with remarkable consistency between samples (Fig~\ref{fig:uncertainty}A).
One explanation for this is that there is much less data per window than for the whole chloroplast, so estimates at any individual window will have larger standard errors, inflating the variance between windows.
However, the observed variance between windows was 6.4-fold greater than between simulated draws from a binomial distribution using the observed coverage at each window as the number of trial and the global mean non-conversion rate as the mean.
Moreover, Spearman correlation coefficients ($\rho$) between replicates ranged from 0.71 to 0.84, showing that the pattern was remarkably consistent between technical replicates of the library preparation protocol (Fig~\ref{fig:uncertainty}A).
We repeated this analysis for windows of 11~bp, corresponding to one helix turn around a histone, and found similar results.
These observations indicate that the variation in non-conversion across the chloroplast reflects something intrinsic about the DNA rather than stochastic variation.

\begin{figure*}
  \begin{center}
    \includegraphics{figure2.eps}
    \caption{
        {\bf Non-conversion errors vary at the kilobase scale.}
        Cytosine non-conversion across the first half of the chloroplast in four independent replicate library preparations of the \emph{A.~thaliana} accession Col-0.
        Colours denote independent libraries.
        (A) 150~bp windows across the chloroplast.
        (B) Relationship between non-conversion and coverage per base.
        (C) Relationship between non-conversion and GC content.
    }
    \label{fig:uncertainty}
  \end{center}
\end{figure*}

What might explain this variation?
There was a weak negative correlation ($\rho$ between -0.29 and -0.10) between non-conversion and coverage at each window in three samples, with a weak positive correlation in the remaining sample ($\rho = 0.18$; Fig~\ref{fig:uncertainty}C).
Nevertheless, while very low coverage causes estimates to be noisy, the lowest coverage per cytosine in any window was 32.7x, so this cannot account for a large proportion of the variance.
There were also consistent positive correlations between non-conversion and GC-content in each window across samples ($\rho$ between 0.43 and 0.55; Fig~\ref{fig:uncertainty}C).
The density of CG, CHG and CHH sites explained 1.7\%, 9.9\% and 10.6\% of the variation in (logit) non-conversion rate respectively.
Since CG are the least, and CHH the most common sequence motifs, this is consistent with a parsimonious explanation that the overall density of CG sites predict the overall correlation with non-conversion rates.
In summary, it appears that non-conversion errors vary between regions of the genome in a systematic way.
Although the underlying reasons for this are unclear, this observation warrants caution in using a single point estimate to account for errors across the genome.

\subsection*{Errors are consistent with Tn5 overactivity}

Summarising results so far, our observations indicate that non-conversion errors occur at high frequency in samples from diverse organisms, but that error rates vary between reads and across the genome.
It appears that affected reads result from two distinct processes that generate completely and partially unconverted reads, with a bias towards non-conversion of cytosines towards the 3` ends of reads.
These observations are consistent with the hypothesis that the Tn5 transposase creates single-strand nicks, which are repaired by strand displacement by the \textit{Bst} polymerase in the presence of 5-methyl-dCTPs.
Lu \textit{et al.} \cite{lu2015improved} hypothesised that Tn5 nicks the adaptor sequence, leading to displacement of the entire fragment.
This model would account for those reads where every cytosine is unconverted (Fig~\ref{fig:reads}B).
We speculate that the partially non-converted reads are due a second nicking process that occurs randomly throughout the body of a DNA fragment.
Because the \textit{Bst} polymerase replicates in 5\`{} to 3\`{} direction, random nick positions would systematically bias non-conversion towards downstream cytosines.
This is consistent with the observation that non-conversion errors are more common towards the 3` end of reads (Fig~\ref{fig:reads}C).
Tn5 is known to preferentially insert at guanine and cytosine positions \cite{goryshin1998tn, green2012insertion},
which is consistent with the correlation between non-conversion rates and GC content (Fig~\ref{fig:uncertainty}C).
Moreover, there is evidence that Tn5 insertion sites tend to be spaced at 5-bp intervals \cite{goryshin1998tn}, which matches the periodicity in non-conversion errors in \textit{D.~melanogaster} (Fig \ref{fig:reads}C).
However, it is not known whether patterns of insertion site bias reflect the same process as a bias in the location of single-strand nicks.
Although these data do not allow us to say with certainty what is causing non-conversion errors, this model of the combined action of single-strand nicks followed by strand displacement seems to be a reasonable working hypothesis to explain non-conversion errors.

It is worth considering the implications of this model for other applications.
The fact that strand-displacement protocols use methylated dCTPs but oligo-replacement protocols use unmethylated dCTPs provides a straightforward explanation for why the former have substantial conversion errors while the latter apparently do not.
However, both approaches involve Tn5 transposase, and it stands to reason that single-strand breaks should occur with similar frequency in both.
If these are repaired with unmethylated dCTPs this would introduce a bias in the opposite direction to that described here that would \textit{reduce} apparent methylation.
Importantly, such a bias would almost certainly go undetected unless a completely methylated control sequence were included.
If our working hypothesis is correct, it is unlikely that any such bias would be of the same magnitude as the non-conversion errors we describe because oligo-replacement protocols do not involve strand displacement.
Nevertheless we suggest that some caution is warranted regarding possible false-conversion errors in standard tagmentation protocols.

How can we deal with non-conversion errors in the light of this working hypothesis?
Suzuki \textit{et al.}\cite{suzuki2018whole} filtered reads if they have multiple unconverted CHH sites (H is any nucleotide except G).
This makes sense in mammals where methylation is primarily on CG sites, but will not work for plants where CHH methylation is both real and interesting.
Lu \textit{et al.} \cite{lu2015improved} scored reads based on the number and position of unconverted cytosines, and discard suspicious reads.
However, the logic of this score is not clear, and was developed with completely non-converted reads in mind.
It will likely not work in the presence of partially unconverted reads.
An alternative partial fix would be to filter out completely unconverted reads, but this would not address the partially unconverted reads, and risks biasing inferred methylation in regions that are truly highly methylated.
Fortunately, since non-conversion errors happen in vitro and not in vivo we can account for them by modelling the error-generating process statistically.
In the sections that follow and the Materials and Methods we describe a framework to model non-conversion errors and apply it to experimental data from \emph{A.~thaliana}.

\subsection*{Methylation estimates depend more on sample size than non-conversion errors}

Often one may wish to estimate and compare average methylation within and between regions of the genome.
Estimates of average methylation reflect a binomial sampling process of converted and unconverted cytosines from different cells.
We aim to estimate this mean with as little bias and sampling variance as possible.
Non-conversion of truly unmethylated cytosines will bias apparent methylation upwards.
Likewise, incorrect conversion of truly methylated cytosines will bias methylation downwards.
In this study we focus on the former prcoess, but note that the latter is distinctly plausible.
Since these errors are statistical rather than biological, we can account for them by allowing for the probabilities of errors in the binomial sampling process (see Methods).
Given a good estimate of error rates, it is straightforward adjust estimates to correct these biases.

\begin{figure*}
  \begin{center}
    \includegraphics{figure3.eps}
    \caption{
        {\bf Estimates of mean methylation in simulated data using different estimates of the non-conversion error rate.}
        Plots show the effect of combining data over 1, 10 or 100 loci with 200 cytosines each for increasing sequencing depth.
        True methylation is 0.15 for all loci.
        Error rates for each locus are either zero or are drawn from a beta distribution with shape parameters $a=17$ and $b=220$.
        Methylation level is estimated for data with no errors, for data with errors using the true error rate for the simulation, and for data with errors when the error rate is estimated as the mean of the beta distribution.
        (A) Distribution of estimated means across replicate 200 simulations.
        (B) Mean deviation from the true value across simulations.
        Note that points are offset along the x-axis for visual clarity.
        (C) Mean bias across simulations, calculated as the observed estimated methylation level minus true methylation level.
    }
    \label{fig:simulations}
  \end{center}
\end{figure*}

However, this correction is complicated by two practical concerns that effectively increase sample variance.
First, as noted above, there seems to be substantial variation in error rates between windows of the chloroplast.
If we try to use the chloroplast to gauge error rates at the rest of the genome, this implies that there is substantial uncertainty around the error rate at any specific window.
This will tend to overestimate true methylation when the error rate is above the mean error rate, and vice versa.
Second, the binomial sampling process itself introduces uncertainty about true methylation level.
This is especially true when the number of 'binomial trials' is small, such as when only a small region of the genome is examined and/or when sequencing coverage is low.
Variation in error rates and sample size should be expected to hinder our ability to estimate mean methylation accurately.

We used simulations to investigate the practical significance of conversion errors and binomial sampling on estimates of mean methylation.
We simulated converted and unconverted reads on transposable elements (TEs) of typical size with and without non-conversion errors, and attempt to recover true methylation levels.
We also summed reads across multiple loci, each with a unique error rate, to reflect summing over multiple TEs in the same region or family (e.g. \cite{sasaki2019common}).
When data contain no non-conversion errors (red points in Figs ~\ref{fig:simulations}A and \ref{fig:simulations}B) any deviation from the true mean reflects variance due to binomial sampling only, and represents the best-case scenario for given coverage and number of loci.
When non-conversion errors are introduced but the error rate is known we observe an increase in variation around the true mean (green).
This additional error must reflect the additional binomial variance that arises from sampling from both truly methylated and unmethylated reads.
When the error rate is not known but is estimated (blue) estimated means are even less accurate.
This demonstrates that both binomial sampling and uncertainty in the true error rate contribute to uncertainty in true methylation rate.

Despite this, the effect of error rates on accuracy was much smaller than the effect of sample size.
The accuracy of estimated means increased dramatically with increased number of trials, either by increasing the number of loci or sequence coverage, including when data are simulated without errors (Figs~\ref{fig:simulations}A and \ref{fig:simulations}B).
This is not surprising in itself, but it is notable that a fairly simple correction for error rates can recover good estimates of mean methylation over thousands of cytosines using modest sequence coverage, even with relatively high non-conversion errors.
This implies that the most practical way to ensure accurate estimates of methylation is to increase sample size, either by increasing sequencing depth or by summing over larger windows or multiple loci.
Since sequencing depth is typically limited by budgets, the latter option is likely the most practical, and often makes biological sense.
For example, multiple TEs may be targeted and co-regulated by the same silencing pathways, so it may be logical to examine them together.
Increasing the number of binomial trials is the surest path to accurate estimates of mean methylation, even when errors are common.

Aside from any effects on the variance, cytosine non-conversion errors should systematically increase the apparent number of methylated cytosines, and bias estimates of methylation level upwards if ignored.
However, our estimates of mean methylation level after adjustment for non-conversion errors are symmetrical around the true value (Fig~\ref{fig:simulations}A) and show no evidence for systematic bias (Fig~\ref{fig:simulations}C).
Moreover, the scale of any bias is an order magnitude less than that of overall deviations (Figs~\ref{fig:simulations}B and \ref{fig:simulations}C).
These observations are true even when there is uncertainty about the true error rate.
This indicates that the statistical correction for non-conversion errors is effective in removing the bias, and that sampling variance in estimated methylation level is a greater concern than bias caused by non-conversion errors.

\subsection*{Imputation of methylation state}

In some applications one might wish to classify a discrete methylation `state', rather than quantify methylation level.
For example, if the biological question addresses whether a locus is targeted by silencing machinery or not it may be practical to classify a region as `methylated' or `unmethylated', rather than try to deal with noisy or overdispersed quantitative estimates.
To illustrate this we categorised methylation status of regions of the \emph{A.~thaliana} genome based on a model of methylation homeostasis posited by Zhang \textit{et al.} \cite{zhang2020natural}.
In plants, cytosines can be methlylated in all three sequence contexts (CG, CHG and CHH) maintained by distinct pathways \cite{law2010establishing}.
Although much of the \emph{A.~thaliana} genome is unmethylated, some genes display `gene-body methylation' characterised by methylation of CG sites, whose origin and function has attracted much attention \cite{muyle2022gene}. 
Transposable elements are associated with methylation in all three of these contexts \cite{cokus2008shotgun, lister2008highly}.
Zhang \textit{et al.} \cite{zhang2020natural} describe a continuum from unmethylated to gene-body-like methylation (GBM-like) to TE-like methylation.
Since these three states are likely maintained by distinct biological pathways it makes sense to model them as distinct methylation states.

\begin{figure*}
    \centering
    \includegraphics{figure4.eps}
    \caption{
        {\bf Methylation state calls in windows of the chloroplast, genes, CMT-targetted TEs and RdDM-targetted TEs.}
        Plots show the posterior probabilities that a single window/gene/TE is in each state.
        Only those features with at least one aligned read in each sequence context are shown.
    }
    \label{fig:meth-state}
\end{figure*}

We categorised methylation status for bisulphite data in the presence of substantial non-conversion errors in our \emph{A.~thaliana} samples.
We used a model that compares the evidence that the counts of observed converted and unconverted reads in the three sequence contexts are consistent with non-conversion errors only, or a mixture of errors and real cytosine methylation, corresponding to truly zero and non-zero methylation levels respectively.
We applied this model to (1) windows of the chloroplast, which should be unmethylated; (2) annotated transposable elements known to be actively regulated by the CMT2 and RdDM pathways \cite{stroud2013comprehensive} and likely to be TE-like methylated; (3) coding genes, likely reflecting a mix of all three methylation states \cite{zhang2020natural}.
For each window/locus we quantified the evidence that DNA was unmethylated, GBM-like methylated, or TE-like methylated.

The observed states of the chloroplast and TEs showed a strong tendency towards their expected state (Fig~\ref{fig:meth-state}).
95.6\% of windows on the chloroplast were most likely to be unmethylated.
Likewise, 94.9\% of CMT2-targetted TEs and 86.9\% of RdDM-targettted TEs were TE-like methylated.
We found nearly identical results when we sum over the probabilities that each window/TE is in each state rather than make hard calls about methylation state (Fig~\ref{fig:meth-state}).
The methylation state of unmethylated DNA and transposable elements can be inferred despite non-conversion errors.

Methylation states of genes were also qualitatively similar to previous reports.
47.8\% of genes were most likely to be unmethylated, 36.9\% GBM-like methylated and 15.3\% TE-like methylated.
This matches the general pattern of methylation states reported by Zhang \textit{et al.} \cite{zhang2020natural}, although that study inferred approximately fifty percent more unmethylated genes and around half as many GBM-like and TE-like genes.
In contrast to the chloroplast and TEs, it is not clear what the expectation should be, and gene methylation states likely reflect a dynamic equilibrium \cite{zhang2020natural}.
Non-conversion errors in these data almost certainly contribute this quantitative discrepancy, but they are also likely driven to a large part by differences in how methylation states are called.
States inferred by Zhang \textit{et al.} \cite{zhang2020natural} were based on an approach described by Takuno \& Gaut \cite{takuno2012body} that compares methylation at each gene to the genome-wide average via a binomial test.
This test is conservative, in that only the most strongly gene-body methylated genes will show a significant increase in methylation, which will tend to increase the apparent incidence of unmethylated genes.
Moreover, it is not clear that comparing methylation at individual loci to a genome-wide average is an appropriate null hypothesis.
For example, if there were a genetic or environmental modifier that increased global methylation it would in fact be \textit{more} difficult to identify individual loci with strong methylation.
In contrast, our approach quantifies the evidence for competing hypotheses (possible methylation states) based on the evidence available for each locus on its own \cite{chamberlin1890method, burnham2004multimodel}.

\section*{Conclusions}

Strand-displacement-based tagmentation protocols allow for bisulphite sequencing of samples with smaller amounts of starting material, and are more easily scalable to high-throughput samples than previous protocols.
However, this scalability comes at the cost of substantial cytosine non-conversion errors.
The patterns of these errors is consistent the Tn5 transposase making single-strand nicks across fragments, followed by strand displacement of the rest of the downstream fragment by \textit{Bst} polymerase, replacing every cytosine with 5-methyl-dCTPs.
Fortunately, these errors can be modelled and corrected statistically.
We demonstrate that non-conversion errors have less of an effect on quantitative estimates of methylation than does binomial sampling, which is common to all protocols.
Ultimately the additional work in dealing with errors should be traded off against the ease of scalability of strand-displacement-based bisulphite protocols.

\section*{Materials and Methods}

\subsection*{Bisulphite sequencing}

We performed bisulphite sequencing of DNA from \textit{A.~thaliana}, \emph{D.~melanogaster} and phage $\lambda$.
We extracted DNA from the \textit{A.~thaliana} accession Col-0 using the NucleoMag 96 Plant kit (Macherey Nagel Bioanalysis REF. 744400.24).
An aliquot of \textit{D.~melanogaster} DNA extracted from ovarian somatic cell culture \cite{niki2006establishment, saito2009regulatory} was kindly donated by Dominik Handler.
We prepared bisulphite libraries based on the protocol described by Weichenhan \textit{et al.} \cite{weichenhan2019generation} with some modifications.
We used 20-30 ng of genomic DNA per sample.
As an internal control, we added 5 pg of unmethylated DNA from phage $\lambda$ (New England Biolabs, E7123A) per 10 ng sample DNA.
We performed tagmentation using in-house Tn5 transposase \cite{picelli2014tn5} in ratios of 1:1 and 1:2 transposase to DNA.
We used EZ-96 DNA Methylation-Gold MagPrep (Zymo, D5043) for bisulphite conversion.
We did 13 or 15 cycles of PCR amplification for enrichment, and we used Illumina Nextera DNA indexes (i7/i5) to allow multiplexing.
We performed size selection and PCR clean-up with in-house SPRI beads.
We validated libraries by Fragment Analyzer™ Automated CE System (Advanced Analytical) and pooled them in equimolar concentration.
Median insert size was around 350 bp.

Libraries were sequenced by the Next Generation Sequencing Facility at Vienna BioCenter Core Facilities (VBCF), on an Illumina NovaSeq Analyzer using manufacturer’s standard cluster generation ad sequencing tools (S2 flowcell with 100 bp paired end reads).
To compensate for low nucleotide diversity in bisulphite libraries, each lane included a 15\% splike in of phiX Control.

\subsection*{Bioinformatics}

We used cutadapt to remove adaptor sequences from raw reads \cite{martin2011cutadapt}.
We also trimmed 15 bp from the 5` and 9 bp from the 3` end of each read corresponding to the gap-repair regions of each fragment and visually inspected the quallity of unaligned reads using FastQC \cite{andrews2020fastqc}.
We used Bismark to align reads to the TAIR10 and \textit{D.~melanogaster} 6 reference genomes \cite{krueger2011bismark}.
We concatenated both reference genomes with the Lambda phage genome.
In addition, we included the genomes of the microbial symbionts \textit{Acetobacter aceti} and \textit{Lactobacillus acidophilus}.
We used Bismark to remove PCR-duplicated sequences and calculate the number of converted and unconverted cytosines aligning to each genomic cytosine position, and used the resulting cytosine report files to quantify overall non-conversion rates across the Drosophila and Lambda-phage genomes.

\subsection*{Data analysis}

We used custom Python functions to quantify patterns of non-conversion within reads and across the genome, available via the package \textit{methlab} \cite{ellis2023methlab}. For \emph{A.~thaliana} we estimated non-conversion on the first half of the chloroplast only, because the second half shows strong homology with one of the autosomes.

We carried out statistical analyses in \textit{R} 4.3.3 using \textit{RStudio} 2023.06.1 \cite{rlanguage, rstudio} and \textit{ggplot2} for plotting \cite{wickham2011ggplot2}. To investigate patterns of non-conversion rates between windows of the \textit{A.~thaliana} chloroplast we modelled (logit) non-conversion rate with the numbers of CG, CHG and CHH sites as fixed effects, and a random effect of library ID as a random effect using \textit{lme4} \cite{bates2015}. The logit transformation accounts for the non-linearity of variance that occurs when data are bounded by zero and one. We used the R package \textit{r2glmm} 0.1.2 to estimate the variance explained by each fixed effect \cite{r2glmm}.

The computational results presented were obtained using the CLIP cluster (\url{https://clip.science}).

\subsection*{Statistical models to account for conversion errors}

\subsubsection*{Error rates in a binomial model} \label{sec:binomial-with-errors}

Given an alignment of bisulphite-converted reads to a genome we observe a total of $n$ reads mapping to genomics cytosines within a given region.
Of these, $y$ reads are observed to be unconverted cytosines and $n-y$ reads appear as converted thymines.
The goal is to estimate the true mean methylation level $\theta$ which generated these data, accounting for conversion errors.

In the absence of errors, the likelihood of the data given $\theta$ is binomially distributed as
\begin{equation}
    \label{eqn:classic-binomial}
    \Pr(y | \theta) = \binom{n}{y} \theta^y(1-\theta)^{n-y}
\end{equation}
with mean $\theta=y/n$.
Data are not perfect, so we would like to incorporate two error terms:
\begin{itemize}
    \item $\epsilon_1$ is the probability that a truly unmethylated cytosine appears methylated (the bisulphite non-conversion rate).
    \item $\epsilon_2$ is the probability that a truly methylated cytosine appears unmethylated, which we include for completeness.
\end{itemize}
Reads observed as cytosines may thus be either truly methylated with probability $\theta(1-\epsilon_2)$ or truly unmethylated with probability $(1-\theta)\epsilon_1$. Likewise, reads observed as thymine may be either truly unmethylated with probability $(1-\theta)(1-\epsilon_1)$ or truly methylated with probability $\theta \epsilon_2$.
This changes the likelihood to

\begin{equation}
\begin{split}
    \label{eqn:binom-with-errors}
    \Pr(y | \theta, \epsilon_1, \epsilon_2) = 
    {n \choose y}
    [\theta(1-\epsilon_2) + (1-\theta)\epsilon_1]^y \\
    [\theta \epsilon_2 + (1-\theta)(1-\epsilon_1)]^{n-y}
\end{split}
\end{equation}

Summarising $p=[\theta(1-\epsilon_2) + (1-\theta)\epsilon_1] = y/n$ for brevity, this has a closed-form maximum-likelihood estimate
\begin{equation}
    \label{eqn:ml-theta}
    \hat{\theta} = \frac{\epsilon_1-p}{\epsilon_1 + \epsilon_2 -1}
\end{equation}
Note that $\hat{\theta}<0$ when $\epsilon_1 > \theta$ and $\hat{\theta}>1$ when $\epsilon_2 > (1-\theta)$, which are not valid results.
In these cases we set $\hat{\theta}$ to zero and one respectively.

\subsubsection*{Accounting for uncertainty in error rate} \label{sec:mean-as-beta}

The above formulation is valid for the case where there is a clear point estimate for error rates.
If there is substantial uncertainty around these estimates, and especially if they are likely to vary across the genomes, then error rates can be modelled as coming from beta distributions with shape parameters $a$ and $b$. For DNA which is known to be unmethylated, such that all unconverted cytosines are incorrect:
\begin{equation}
    \label{eqn:beta-dist}
    \Pr(p|a,b) = \frac{p^{a-1} (1-p)^{b-1}}{\textrm{B}(a,b)}
\end{equation}
where $\textrm{B}(a,b)$ is the Beta function.
The shape parameters can be estimated by method-of-moments using estimates of the sample mean $\mu$ and variance $\sigma$ of the distribution.
For example, one can calculate mean ($\mu$) and variance ($\sigma^2$) in the proportion of unconverted cytosines across windows of DNA known to be unmethylated.
Estimates for shape parameters are then:
\begin{equation}
    \hat{a} = \mu(\frac{\mu(1-\mu)}{\sigma^2}-1)
    \label{eqn:beta-a}
\end{equation}
\begin{equation}
    \hat{b} = (1-\mu)(\frac{\mu(1-\mu)}{\sigma^2}-1) 
    \label{eqn:beta-b}
\end{equation}

\subsubsection*{Classification of methylation state} \label{sec:meth-state}

In some cases it makes biological sense to classify the 'state' of a region as methylated or unmethylated rather than quantify methylation level.
If we assume false conversion errors ($\epsilon_2$) are negligible, this can be achieved by comparing the evidence that the proportion of unconverted cytosines $p$ is drawn from distribution $f(p)$ generating false non-conversion errors, or from an alternative distribution $g(p)$ that reflects real biological processes.

First, $f(p)$ can be described as a combination of the binomial and beta processes described above (Eqs~\ref{eqn:binom-with-errors} and \ref{eqn:beta-dist}).
If all the reads are truly unmethylated then $\theta=0$, and hence
\begin{equation}
    f(p) = \Pr(y | p, n) \Pr(p | a, b)
    \label{eqn:f-of-p}
\end{equation}
where $a, b$ are shape parameters of the beta distribution of error rates.
This accounts for both the binomial process of drawing a finite sample of reads, and the uncertainty in error rate.

It may be possible to modify Eq~\ref{eqn:f-of-p} to completely describe the distribution of truly methylated cytosines.
In this case $\epsilon_1=\epsilon_2=0$, but $\theta>0$, and hence
\begin{equation}
    g(p) = \Pr(y | p, n) \Pr(p | a^*,b^*)
\end{equation}
where $a^*$ and $b^*$ are the beta shape parameters for the distribution of true methylation levels.
This is only valid if it is possible to estimate these shape parameters, for example from external data from libraries prepared using a different protocol for which conversion error rates are negligible.

However, as long as the true methylation level is substantially larger than error rates, we can simplify the problem by asking whether observed unconverted reads are compatible with non-conversion errors alone, or with more methylation than that.
A simpler approach is to model 
\begin{equation}
    g(p) = \Pr(y | p, n) \Pr(p \leq P | a,b)
\end{equation}
where $\Pr(p \leq P | a,b)$ is the cumulative density function of the beta distribution of error rates.
This approaches zero when the proportion of unconverted reads is small, and approaches one when the number of unconverted reads is more than can be accounted for by $f(p)$, and thus readily distinguishes 'high' and 'low' non-conversion rates.
This has the advantage that we need only know $a$ and $b$, which can be estimated in the presence of cytosine non-conversion errors.
For this reason, we use this estimate of $g(p)$ in this manuscript.

We can apply this model to quantify the evidence that a region of DNA is unmethylated (um), gene-body-like methylated (gbm) or TE-like methylated (TEm) based on the observed proportions of non-converted reads in the CG, CHG and CHH sequence contexts ($p_{CG}$, $p_{CHG}$ and $p_{CHH}$; H is any nucleotide apart from G).
In unmethylated DNA all unconverted cytosines are due to non-conversion errors, and hence the likelihood that the region is unmethylated is
\begin{equation}
    L_{um} = f(p_{CG}) f(p_{CHG}) f(p_{CHH})
\end{equation}
In gene-body-like-methylated DNA, CG sites are methylated but CHG and CHH sites are not: 
\begin{equation}
    L_{gbm} = g(p_{CG}) f(p_{CHG}) f(p_{CHH})
\end{equation}
In TE-like-methylated DNA cytosines are methylated in all contexts:
\begin{equation}
    L_{TEm} = g(p_{CG}) g(p_{CHG}) g(p_{CHH})
\end{equation}

These likelihoods can be converted to probabilities by dividing by the sum of the three likelihoods.
The global probability that any single locus is in methylation state $i$, accounting for the uncertainty in state at each locus, is simply the average probability of being in state $i$ over all loci.

\subsection*{Simulations}

We used simulations to investigate how well we can estimate true mean methylation when observed data are contaminated with non-conversion errors.
We simulated loci of 800 nucleotides (200 cytosines), corresponding to the average size (798bp) of a transposable element or TE fragment in the \textit{A.~thaliana} TAIR10 annotation, with 1, 2, 4, 8, 16 or 32 reads mapping to each cytosine.
We simulated the conversion state of each read at each cytosine as a binomial process with the total number of trials being the number of cytosines multiplied by coverage per cytosine.
Following Eq~\ref{eqn:binom-with-errors} and ignoring false conversion errors, each read may be observed as unconverted with probability $p=\theta + (1-\theta)\epsilon_1$, where $\theta$ is the probability that the read is truly methylated and $\epsilon_1$ is a non-conversion error rate.
We used $\theta=0.15$ for these simulations, which is both a realistic estimate of TE methylation level \cite{dubin2015dna}.
We note that much higher methylation levels would be expected for CpG islands or gene-bodies; we use the lower value because this represents the more challenging case where real methylation levels are close to non-conversion error rates.

For each locus we drew a value for $\epsilon_1$ from a beta distribution describing variation in error rates across the genome.
To estimate biologically realistic shape parameters for this distribution we estimated the mean and variation in non-conversion rates in 150 bp windows across the chloroplast for one of the \emph{A.~thaliana} samples, and calculated beta shape parameters by method of moments (Eqs~\ref{eqn:beta-a} and \ref{eqn:beta-b}).

\section*{Data and code availability}

Sequencing data generated in this study are available in the National Center for Biotechnology Information (NCBI) Sequence Read Archive (SRA) under the accession number PRJNA1155267 \cite{ellis2023datafrom}.
Code to run the analyses is available from Github (\url{https://github.com/ellisztamas/bs_seq_with_typing_errors}) and Zenodo \cite{ellis2023codefrom}.

\section*{Author contributions}

TJE: data curation, processing and analysis, study design, writing.
VN: study design, preparation of sequencing libraries.
RP: data processing.
AMM: study design, coordination of resources, critically revised the manuscript.
MN: supervision of the project, critically revised the manuscript.

\section*{Acknowledgements}

We thank Robert Schmitz, Dieter Weichenhan and members of the Nordborg group for discussion of the results.
We specifically thank Yoav Voichek for technical assistance on how to disembowel SAM files.
We also thank Dominik Handler for donated an aliquot of \emph{D.~melanogaster} DNA.

\nolinenumbers

% \bibliography{04_manuscript/refs}

\begin{thebibliography}{10}

\bibitem{clark1994high}
Clark SJ, Harrison J, Paul CL, Frommer M.
\newblock High sensitivity mapping of methylated cytosines.
\newblock Nucleic acids research. 1994;22(15):2990--2997.

\bibitem{meissner2005reduced}
Meissner A, Gnirke A, Bell GW, Ramsahoye B, Lander ES, Jaenisch R.
\newblock Reduced representation bisulfite sequencing for comparative high-resolution DNA methylation analysis.
\newblock Nucleic acids research. 2005;33(18):5868--5877.

\bibitem{cokus2008shotgun}
Cokus SJ, Feng S, Zhang X, Chen Z, Merriman B, Haudenschild CD, et~al.
\newblock Shotgun bisulphite sequencing of the Arabidopsis genome reveals DNA methylation patterning.
\newblock Nature. 2008;452(7184):215--219.

\bibitem{lister2009human}
Lister R, Pelizzola M, Dowen RH, Hawkins RD, Hon G, Tonti-Filippini J, et~al.
\newblock Human DNA methylomes at base resolution show widespread epigenomic differences.
\newblock nature. 2009;462(7271):315--322.

\bibitem{wang2013tagmentation}
Wang Q, Gu L, Adey A, Radlwimmer B, Wang W, Hovestadt V, et~al.
\newblock Tagmentation-based whole-genome bisulfite sequencing.
\newblock Nature protocols. 2013;8(10):2022--2032.

\bibitem{adey2012ultra}
Adey A, Shendure J.
\newblock Ultra-low-input, tagmentation-based whole-genome bisulfite sequencing.
\newblock Genome research. 2012;22(6):1139--1143.

\bibitem{lu2015improved}
Lu H, Yuan Z, Tan T, Wang J, Zhang J, Luo HJ, et~al.
\newblock Improved tagmentation-based whole-genome bisulfite sequencing for input DNA from less than 100 mammalian cells.
\newblock Epigenomics. 2015;7(1):47--56.

\bibitem{weichenhan2019generation}
Weichenhan D, Imbusch CD, Wang Q, Brors B, Plass C.
\newblock Generation of Whole Genome Bisulfite Sequencing Libraries from Very Low DNA Input.
\newblock Lymphoma: Methods and protocols. 2019; p. 229--248.

\bibitem{suzuki2018whole}
Suzuki M, Liao W, Wos F, Johnston AD, DeGrazia J, Ishii J, et~al.
\newblock Whole-genome bisulfite sequencing with improved accuracy and cost.
\newblock Genome research. 2018;28(9):1364--1371.

\bibitem{zhang2018large}
Zhang Y, Harris CJ, Liu Q, Liu W, Ausin I, Long Y, et~al.
\newblock Large-scale comparative epigenomics reveals hierarchical regulation of non-CG methylation in \textit{Arabidopsis}.
\newblock Proceedings of the National Academy of Sciences. 2018;115(5):E1069--E1074.

\bibitem{goryshin1998tn}
Goryshin IY, Miller JA, Kil YV, Lanzov VA, Reznikoff WS.
\newblock Tn 5/IS 50 target recognition.
\newblock Proceedings of the National Academy of Sciences. 1998;95(18):10716--10721.

\bibitem{green2012insertion}
Green B, Bouchier C, Fairhead C, Craig NL, Cormack BP.
\newblock Insertion site preference of Mu, Tn5, and Tn7 transposons.
\newblock Mobile DNA. 2012;3:1--6.

\bibitem{sasaki2019common}
Sasaki E, Kawakatsu T, Ecker JR, Nordborg M.
\newblock Common alleles of CMT2 and NRPE1 are major determinants of CHH methylation variation in \emph{Arabidopsis thaliana}.
\newblock PLoS Genetics. 2019;15(12):e1008492.

\bibitem{zhang2020natural}
Zhang Y, Wendte JM, Ji L, Schmitz RJ.
\newblock Natural variation in DNA methylation homeostasis and the emergence of epialleles.
\newblock Proceedings of the National Academy of Sciences. 2020;117(9):4874--4884.

\bibitem{law2010establishing}
Law JA, Jacobsen SE.
\newblock Establishing, maintaining and modifying DNA methylation patterns in plants and animals.
\newblock Nature Reviews Genetics. 2010;11(3):204--220.

\bibitem{muyle2022gene}
Muyle AM, Seymour DK, Lv Y, Huettel B, Gaut BS.
\newblock Gene body methylation in plants: mechanisms, functions, and important implications for understanding evolutionary processes.
\newblock Genome Biology and Evolution. 2022;14(4):evac038.

\bibitem{lister2008highly}
Lister R, O'Malley RC, Tonti-Filippini J, Gregory BD, Berry CC, Millar AH, et~al.
\newblock Highly integrated single-base resolution maps of the epigenome in Arabidopsis.
\newblock Cell. 2008;133(3):523--536.

\bibitem{stroud2013comprehensive}
Stroud H, Greenberg MV, Feng S, Bernatavichute YV, Jacobsen SE.
\newblock Comprehensive analysis of silencing mutants reveals complex regulation of the Arabidopsis methylome.
\newblock Cell. 2013;152(1):352--364.

\bibitem{takuno2012body}
Takuno S, Gaut BS.
\newblock Body-methylated genes in \emph{Arabidopsis thaliana} are functionally important and evolve slowly.
\newblock Molecular Biology and Evolution. 2012;29(1):219--227.

\bibitem{chamberlin1890method}
Chamberlin TC.
\newblock The method of multiple working hypotheses.
\newblock Science. 1890;(366):92--96.

\bibitem{burnham2004multimodel}
Burnham KP, Anderson DR.
\newblock Multimodel inference: understanding AIC and BIC in model selection.
\newblock Sociological methods \& research. 2004;33(2):261--304.

\bibitem{niki2006establishment}
Niki Y, Yamaguchi T, Mahowald AP.
\newblock Establishment of stable cell lines of \emph{Drosophila} germ-line stem cells.
\newblock Proceedings of the National Academy of Sciences. 2006;103(44):16325--16330.

\bibitem{saito2009regulatory}
Saito K, Inagaki S, Mituyama T, Kawamura Y, Ono Y, Sakota E, et~al.
\newblock A regulatory circuit for \textit{piwi} by the large Maf gene \textit{traffic jam} in \textit{Drosophila}.
\newblock Nature. 2009;461(7268):1296--1299.

\bibitem{picelli2014tn5}
Picelli S, Bj{\"o}rklund {\AA}K, Reinius B, Sagasser S, Winberg G, Sandberg R.
\newblock Tn5 transposase and tagmentation procedures for massively scaled sequencing projects.
\newblock Genome research. 2014;24(12):2033--2040.

\bibitem{martin2011cutadapt}
Martin M.
\newblock Cutadapt removes adapter sequences from high-throughput sequencing reads.
\newblock EMBnet journal. 2011;17(1):10--12.
\newblock doi:{10.14806/ej.17.1.200}.

\bibitem{andrews2020fastqc}
Andrews S, Krueger F, Segonds-Pichon A, Biggins L, Krueger C, Wingett S. FastQC: a quality control tool for high throughput sequence data; 2010.
\newblock Available from: \url{https://www.bioinformatics.babraham.ac.uk/projects/fastqc/}.

\bibitem{krueger2011bismark}
Krueger F, Andrews SR.
\newblock Bismark: a flexible aligner and methylation caller for Bisulfite-Seq applications.
\newblock Bioinformatics. 2011;27(11):1571--1572.

\bibitem{ellis2023methlab}
Ellis TJ. methlab: Python tools for the analysis of DNA-methylation data; 2023.
\newblock Available from: \url{https://github.com/ellisztamas/methlab}.

\bibitem{rlanguage}
{R Core Team}. R: A Language and Environment for Statistical Computing; 2024.
\newblock Available from: \url{https://www.R-project.org/}.

\bibitem{rstudio}
{RStudio Team}. RStudio: Integrated Development Environment for R; 2020.
\newblock Available from: \url{http://www.rstudio.com/}.

\bibitem{wickham2011ggplot2}
Wickham H.
\newblock ggplot2.
\newblock Wiley interdisciplinary reviews: computational statistics. 2011;3(2):180--185.

\bibitem{bates2015}
Bates D, M{\"a}chler M, Bolker B, Walker S.
\newblock Fitting Linear Mixed-Effects Models Using {lme4}.
\newblock Journal of Statistical Software. 2015;67(1):1--48.
\newblock doi:{10.18637/jss.v067.i01}.

\bibitem{r2glmm}
Jaeger B. r2glmm: Computes R Squared for Mixed (Multilevel) Models; 2017.
\newblock Available from: \url{https://CRAN.R-project.org/package=r2glmm}.

\bibitem{dubin2015dna}
Dubin MJ, Zhang P, Meng D, Remigereau MS, Osborne EJ, Paolo~Casale F, et~al.
\newblock DNA methylation in Arabidopsis has a genetic basis and shows evidence of local adaptation.
\newblock elife. 2015;4:e05255.

\bibitem{ellis2023datafrom}
Ellis TJ, Nizhynska V, Pisupati R, Moll\'a-Morales A, Nordborg M.
\newblock Data from: Bisulphite sequecing in the presence of cytosine-conversion errors.
\newblock The National Center for Biotechnology Information (NCBI) Sequence Read Archive (SRA); 2023. BioProject: PRJNA1155267.

\bibitem{ellis2023codefrom}
Ellis TJ, Nizhynska V, Pisupati R, Moll\'a-Morales A, Nordborg M. Code from: Bisulphite sequecing in the presence of cytosine-conversion errors; 2025.
\newblock Available from: \url{https://doi.org/10.5281/zenodo.14883552}.

\end{thebibliography}

\end{document}